lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -577,15 +577,15 @@ def set_zsort(self, zsort):
 
     def get_vector(self, segments3d):
         """Optimize points for projection."""
-        if len(segments3d):
-            xs, ys, zs = np.row_stack(segments3d).T
-        else:  # row_stack can't stack zero arrays.
-            xs, ys, zs = [], [], []
-        ones = np.ones(len(xs))
-        self._vec = np.array([xs, ys, zs, ones])
-
-        indices = [0, *np.cumsum([len(segment) for segment in segments3d])]
-        self._segslices = [*map(slice, indices[:-1], indices[1:])]
+        if isinstance(segments3d, (np.ndarray, np.ma.MaskedArray)):
```

Comment:
This is identical to
```suggestion
        if isinstance(segments3d, np.ndarray):
```

Reply:
Perhaps also worth adding
```suggestion
        if isinstance(segments3d, (np.ndarray, np.ma.MaskedArray)):
            if segments3d.ndim != 3 or segments3d.shape[-1] != 3:
                raise ValueError("Wrong shape!")  # or a better message
```
```

Reply:
Second comment here still applies

Reply:
Sorry, it didn't show up when I applied the first suggestion! Should be fixed now.

Reply:
Thanks!